src/csharp/Microsoft.Spark.E2ETest/UdfTests/UdfSimpleTypesTests.cs

```diff
@@ -166,5 +167,37 @@ public void TestUdfWithReturnAsTimestampType()
                 }
             }
         }
+
+        /// <summary>
+        /// UDFs defined in separate threads to validate ThreadStatic bug fix.
+        /// </summary>
+        [Fact]
+        public void TestUdfWithMultipleThreads()
+        {
+            try
+            {
+                static void Method1()
+                {
+                    try
+                    {
+                        Func<Column, Column> udf1 = Udf<string, string>(str => str);
+                    }
+                    catch (Exception e)
+                    {
+                        throw new Exception(e.ToString());
```

Comment:
just rethrow? In that case, do you need this try/catch block? It's doing the same thing.

Reply:
Sorry I fixed these changes but something got missed while merging latest from this branch, please check the latest changes